### PR TITLE
images: Add rhel-10-0

### PR DESCRIPTION
**Pull request by mvollmer**
* [x] image-refresh rhel-10-0

### CODE REVIEW

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-rhel-10-664d9a9a-20240320-150127/log.html

**Comment by mvollmer** [comment]
There are no cloud images yet. Also, the repo URL needs to include "-Public-Beta", if this is indeed the rhel-10 that we are looking for.

**Comment by martinpitt** [comment]
Too early indeed, there's not even c10s images yet either.

**Comment by jelly** [comment]
So the repo urls are a bit funky now:  /rhel-10/nightly/RHEL-10-Public-Beta/latest-RHEL-10/ instead of


rhel-10/nightly/RHEL-10/latest-RHEL-10.0/compose/BaseOS/x86_64/os/repodata/repomd.xml

Do I need to complain somewhere? :)

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-rhel-10-bc003782-20240424-105017/log.html

**Comment by jelly** [comment]
```
+ dnf install -y acl boom-boot clevis-luks cryptsetup dhcp-client dnf-automatic dnf-utils dracut-fips gdb glibc-all-langpacks kmod-kvdo kpatch kpatch-dnf libvirt-daemon-config-network ltrace nmap-ncat python3-tracer qemu-kvm redis socat strace tang targetcli tcsh tlog valgrind vdo virt-install wireguard-tools subscription-manager-cockpit insights-client device-mapper-multipath firewalld glib-networking json-glib kexec-tools libssh libvirt-dbus libvirt-client libvirt-daemon-kvm NetworkManager-team openssl PackageKit pcp-libs pcp podman realmd redhat-logos setroubleshoot-server subscription-manager sos stratis-cli stratisd tuned udisks2 udisks2-lvm2 udisks2-iscsi ipa-client oddjob oddjob-mkhomedir sssd sssd-dbus
Last metadata expiration check: 0:00:02 ago on Wed Apr 24 06:52:01 2024.
Package dhcp-client-12:4.4.3-13.P1.el10.x86_64 is already installed.
Package yum-utils-4.5.0-1.el10.noarch is already installed.
Package dracut-059-22.el10.x86_64 is already installed.
No match for argument: kmod-kvdo
No match for argument: kpatch
No match for argument: kpatch-dnf
No match for argument: vdo
No match for argument: wireguard-tools
Package subscription-manager-cockpit-6-3.el10.noarch is already installed.
Package insights-client-3.2.2-4.el10.noarch is already installed.
Package glib-networking-2.80.0-1.el10.x86_64 is already installed.
Package json-glib-1.8.0-3.el10.x86_64 is already installed.
Package kexec-tools-2.0.28-4.el10.x86_64 is already installed.
Package libssh-0.10.6-4.el10.x86_64 is already installed.
Package NetworkManager-team-1:1.47.5-1.el10.2.x86_64 is already installed.
Package openssl-1:3.2.1-3.el10.x86_64 is already installed.
Package PackageKit-1.2.8-3.el10.x86_64 is already installed.
Package redhat-logos-90.2-2.el10+5.x86_64 is already installed.
Package setroubleshoot-server-3.3.33-1.el10.x86_64 is already installed.
Package subscription-manager-1.29.40-2.el10.x86_64 is already installed.
Package sos-4.6.1-2.el10.noarch is already installed.
Package tuned-2.22.0-0.1.rc1.el10.noarch is already installed.
Package oddjob-0.34.7-12.el10.x86_64 is already installed.
Package oddjob-mkhomedir-0.34.7-12.el10.x86_64 is already installed.
Error: Unable to find a match: kmod-kvdo kpatch kpatch-dnf vdo wireguard-tools
killing ssh master process 131
image-create: setup failed with code 1
```

**Comment by martinpitt** [comment]
@jelly That's known -- we already have a centos-10 special case here: https://github.com/cockpit-project/bots/blob/main/images/scripts/rhel.setup#L234

It needs to be extended to also cover rhel-10.

File: images/scripts/rhel.setup [hunk 1]
```diff
@@ -54,17 +54,23 @@ EOF
         VERSION=${IMAGE#rhel-}
         MAJOR=${VERSION%%-*}
         VERSION=${VERSION/-/.}
+	echo "VERSION=$VERSION"
```
Comment (jelly) [inline images/scripts/rhel.setup:57]: Already dropped locally.

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-rhel-10-365679cf-20240424-145143/log.html

**Comment by martinpitt** [comment]
I resurrect this. We are getting strange errors in RHEL 10 downstream gating, let's give us a way to reproduce/debug them more comfortably.

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-rhel-10-0-17da978f-20240628-092223/log.html

**Comment by cockpituous** [comment]
image-refresh rhel-10-0 done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel-10-0-20240628-095209

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-rhel-10-0-43c9df13-20240628-093547/log.html

**Comment by martinpitt** [comment]
It's happy with starter-kit. I also locally did a `test/image-prepare -q rhel-10-0` in cockpit to try and reproduce the downstream gating failure `TestLogin.testClientCertAuthentication`. It passed. But at the same time that's enough for me to know that this image generally works.

**Comment by martinpitt** [comment]
On the other hand I *can* reproduce the `TestStorageLvm2.testLvmOnLuks` failure. I'll copy the naughty, enable starter-kit in the testmap, then this should be good to go.

**Review by jelly — APPROVED** [review]

**Comment by martinpitt** [comment]
Odd.. the naughty is already present, and it matches:
```
❱❱❱ bots/test-failure-policy -o rhel-10-0 < /tmp/out 
Known issue #5434
```

So this is something else -- let's land this, and I'll enable starter-kit separately.

**Pull request merged**